class.jetpack-options.php

```diff
@@ -134,10 +156,21 @@ public static function get_option( $name, $default = false ) {
 	 * @return mixed|void
 	 */
 	public static function get_option_and_ensure_autoload( $name, $default ) {
-		$value = get_option( $name );
+		$is_network_option = self::is_network_option( $name );
+		if ( $is_network_option ) {
+			$value = get_site_option( $name );
+		}
+		else {
+			$value = get_option( $name );
+		}
 
 		if ( $value === false && $default !== false ) {
-			update_option( $name, $default );
+			if ( $is_network_option ) {
```

Comment:
I'm not sure, but I think you have this conditional backwards. Shouldn't this be `update_site_option` instead?